app/Http/Controllers/GazetteerController.php

```diff
@@ -311,7 +309,11 @@ private function searchDataitems($parameters)
         }
 
         /* BUILD SEARCH QUERY WITH PARAMS */
-        if ($parameters['recordtype']) $dataitems->where('tlcmap.recordtype' . '.type', '=', $parameters['recordtype']);  // Filter by recordtype value
+        if (isset($parameters['recordtype']) && $parameters['recordtype']) {
+            $dataitems->whereHas('recordtype', function ($query) use ($parameters) {
+                $query->where('type', '=', $parameters['recordtype']); // Filter by recordtype value
+            });
+        }

```

Comment:
Can you separate this change along with the one above into a single commit by referencing its related issue number. #63